arvo/eyre/external-api-ref.md

```diff
@@ -0,0 +1,379 @@
+- [Scry](#scry)
+- [Spider](#spider)
+  
+# Introduction
+
+This document contains reference information about Eyre's external APIs including [the channel system](#channels) and [scries](#scry). Each section will also have practical examples in the [Examples](@/docs/arvo/eyre/examples.md) document.
+
+# Authentication
+
+To use Eyre's channel system, run threads or perform scries you must first obtain a session cookie by authenticating with the following HTTP request:
+
+| HTTP Method | Data                                   | URL Path   | Notes                                                                               |
+|-------------|----------------------------------------|------------|-------------------------------------------------------------------------------------|
+| `POST`      | `password=lidlut-tabwed-pillex-ridrup` | `/~/login` | The password is your web login code which can be obtained with `+code` in the dojo. |
+
+Eyre's response will include a `set-cookie` header like:
+
+```
+set-cookie: urbauth-~zod=0v4.ilskp.psv00.t09r0.l8rps.3n97v; Path=/; Max-Age=604800
+```
+
+The `urbauth-{...}` cookie provided must be included in a `cookie` header with all subsequent requests like:
+
+```
+cookie: urbauth-~zod=0v4.ilskp.psv00.t09r0.l8rps.3n97v
+```
+
+# Channels
+
+Eyre's channel system is the primary way of interacting with Gall agents from outside of Urbit.
+
+Data from an external HTTP client is sent to Eyre as an [HTTP PUT request](#http-requests) containing one or more [action](#actions) JSON objects. Clients must obtain a session cookie by [authenticating](#authentication) with Eyre in order to make such requests.
+
+Data from Eyre is sent back to the external HTTP client on a channel as SSEs ([Server Sent Events](https://html.spec.whatwg.org/#server-sent-events)) containing a [response](#responses) JSON object.
+
+A new channel is automatically created whenever a client sends an [action](#action) in an [HTTP PUT request](#http-requests) to `http{s}://{host}{port}/~/channel/{uid}` with a new `uid`. To connect to the channel and receive any pending events, you just send an HTTP GET request with a valid session cookie to that URL.
+
+The response you'll get to the GET request will be an event stream. Events will come in like:
+
+```
+id: 0
+data: {"ok":"ok","id":1,"response":"poke"}
+```
+
+If you're working with Javascript in the browser context you'll handle these with an EventSource object or by using fetch and ReadableStream. If you're using another language, there'll likely be a library available to handle SSEs.
+
+All the events that Eyre sends you on a channel must be [ack](#ack)ed so that Eyre can forget about them and clear them from the channel state. If `fact`s (as [diff](#diff)s) from Gall agents to which you've subscribed are left un`ack`ed long enough, Eyre will consider the particular subscription clogged and automatically unsubscribe you. Note that `Ack`ing one event will implicitly `ack` all previous events.
```

Comment:
```suggestion
All the events that Eyre sends you on a channel must be [ack](#ack)ed so that Eyre can forget about them and clear them from the channel state. If `fact`s (as [diff](#diff)s) from Gall agents to which you've subscribed are left un`ack`ed long enough, Eyre will consider the particular subscription clogged and automatically unsubscribe you. Note that `ack`ing one event will implicitly `ack` all previous events.
```
Not sure if the uppercase Ack was supposed to imply an HTTP header? (I don't really know how HTTP works beyond what I've read here so far!)